### PR TITLE
mednafen: Add version 1.24.1

### PR DESCRIPTION
**Pull request by vhns**
Mednafen is a libre emulator for a few systems.

### CODE REVIEW

File: bucket/mednafen.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+    "version": "1.24.1",
+    "homepage": "https://mednafen.github.io/",
+    "description": "Emulator for different systems",
+    "license": "GNU GPLv2",
```
Comment (Ash258) [inline bucket/mednafen.json:5]:
```suggestion
    "description": "Various systems emulator",
    "homepage": "https://mednafen.github.io/",
    "license": "GPL-2.0-only",
```

File: bucket/mednafen.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            "hash": "d26456eb711feac679e424968edee579b69c38ce1972a93bed10355b04ff21cd"
+        }
+    },
+    "bin": "mednafen.exe",
+    "shortcuts": [
+        [
+            "mednafen.exe",
+            "Mednafen"
+        ]
+    ],
```
Comment (Ash258) [inline bucket/mednafen.json:22]:
```suggestion
    "pre_install": "if (-not (Test-Path \"$persist_dir\\mednafen.cfg\")) { New-Item \"$dir\\mednafen.cfg\" | Out-Null }",
    "bin": "mednafen.exe",
```

File: bucket/mednafen.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "persist": [
+        "cheats",
+        "firmware",
+        "mcm",
+        "mcs",
+        "palettes",
+        "pgconfig",
+        "sav",
+        "snaps"
+        ]
```
Comment (Ash258) [inline bucket/mednafen.json:32]:
```suggestion
        "snaps",
        "mednafen.cfg"
    ],
    "checkver": {
        "url": "https://mednafen.github.io/documentation/ChangeLog.txt",
        "regex": "--\\s+([\\d.]+):\\s+--"
    },
    "autoupdate": {
        "architecture": {
            "64bit": {
                "url": "https://mednafen.github.io/releases/files/mednafen-1.24.1-win64.zip"
            },
            "32bit": {
                "url": "https://mednafen.github.io/releases/files/mednafen-1.24.1-win32.zip"
            }
        },
        "hash": {
            "url": "https://mednafen.github.io/",
            "regex": "(?sm)$basename.*?SHA-256:\\s*$sha256"
        }
    }
```

File: bucket/mednafen.json (hunk 1 above)
Comment (vhns) [reply bucket/mednafen.json:5]: On the license side, I can see that it's probably better in specifying it. But as for the other changes (the order of the homepage and description), even though as I'll merge/commit them anyways, I'd like to know if you have an explanation for it. I supposed it's how all the other packages are written?

File: bucket/mednafen.json (hunk 2 above)
Comment (vhns) [reply bucket/mednafen.json:22]: Thanks a lot for this! I was hesitant on mentioning that I didn't know how to create a persist for the configuration files that aren't within directories.

File: bucket/mednafen.json (hunk 1 above)
Comment (Ash258) [reply bucket/mednafen.json:5]: - Version is required property and main identifier of application. Should be always first.
- Description is the second most valuable information for user who investigate what he will install.
- THen if user wants to know more or the description is not enough, he would go to webpage.
- And finally as soon as he decide to install such application he needs to care about license
	- Corporate statement about some license types
	- License purchase for commercial usage

File: bucket/mednafen.json (hunk 3 above)
Comment (vhns) [reply bucket/mednafen.json:32]: On another note, is there a decided style to be used for the files or is it arbitrary? Do we have something like a clang-format or something to use?

Comment (Ash258) [reply bucket/mednafen.json:32]: What style?

Everything needed is specified editorconfig and validation is done in CI

Comment (Ash258) [reply bucket/mednafen.json:32]: https://ci.appveyor.com/project/lukesampson/scoop-extras/builds/32344679

A

Comment (vhns) [reply bucket/mednafen.json]: Oh, okay. Thank you.

**Comment by pratikpc** [comment]
@vhns could you help us fix #6987?

**Comment by vhns** [comment]
> @vhns could you help us fix #6987?

Unfortunately, no. I no longer use scoop nor Windows.

**Pull request merged**